src/roadrunner-symfony-nyholm/src/Runner.php

```diff
@@ -27,33 +22,12 @@ class Runner implements RunnerInterface
     private $httpMessageFactory;
     private $psrFactory;
 
-    /**
-     * @var array<string, mixed>
-     */
-    private $sessionOptions;
-
-    /**
-     * @param HttpKernelInterface|KernelInterface $kernel
-     */
-    public function __construct($kernel, ?HttpFoundationFactoryInterface $httpFoundationFactory = null, ?HttpMessageFactoryInterface $httpMessageFactory = null)
+    public function __construct(HttpKernelInterface $kernel, ?HttpFoundationFactoryInterface $httpFoundationFactory = null, ?HttpMessageFactoryInterface $httpMessageFactory = null)
     {
         $this->kernel = $kernel;
         $this->psrFactory = new Psr7\Factory\Psr17Factory();
         $this->httpFoundationFactory = $httpFoundationFactory ?? new HttpFoundationFactory();
         $this->httpMessageFactory = $httpMessageFactory ?? new PsrHttpFactory($this->psrFactory, $this->psrFactory, $this->psrFactory, $this->psrFactory);
-
-        if ($kernel instanceof HttpCache) {
```

Comment:
Why isn’t this need anymore?

Reply:
` if ($kernel instanceof HttpCache) {` was only needed to get the paramters from the wrapped kernel. As we are not longer need to handle the sessions here we could remove this. The support for `HttpCache` is still here as `HttpCache` implements `HttpKernelInterface`. Did update the tests which create the Runner instances.